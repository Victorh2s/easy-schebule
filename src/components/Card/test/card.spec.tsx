import '@testing-library/jest-dom';
import { describe, test, expect, vi } from 'vitest';
import { render, fireEvent } from '@testing-library/react';
import { Card } from '../index';

describe('Card component', () => {
  test('renders title correctly', () => {
    const title = 'Test Title';
    const openModalMock = vi.fn();
    const updateTaskMock = vi.fn();
    const deleteTaskMock = vi.fn();

    const { getByText } = render(
      <Card
        title={title}
        openModal={openModalMock}
        updateTask={updateTaskMock}
        deleteTask={deleteTaskMock}
      />
    );
    expect(getByText(title)).toBeInTheDocument();
  });

  test('calls openModal when title is clicked', () => {
    const openModalMock = vi.fn();
    const updateTaskMock = vi.fn();
    const deleteTaskMock = vi.fn();
    const { getByText } = render(
      <Card
        title="Test Title"
        openModal={openModalMock}
        updateTask={updateTaskMock}
        deleteTask={deleteTaskMock}
      />
    );
    fireEvent.click(getByText('Test Title'));
    expect(openModalMock).toHaveBeenCalled();
  });

  test('calls updateTask when Pen icon is clicked', () => {
    const openModalMock = vi.fn();
    const updateTaskMock = vi.fn();
    const deleteTaskMock = vi.fn();
    const { getByLabelText } = render(
      <Card
        title="Test Title"
        openModal={openModalMock}
        updateTask={updateTaskMock}
        deleteTask={deleteTaskMock}
      />
    );
    fireEvent.click(getByLabelText('pen-icon'));
    expect(updateTaskMock).toHaveBeenCalled();
  });

  test('calls deleteTask when Trash icon is clicked', () => {
    const openModalMock = vi.fn();
    const updateTaskMock = vi.fn();
    const deleteTaskMock = vi.fn();
    const { getByLabelText } = render(
      <Card
        title="Test Title"
        openModal={openModalMock}
        updateTask={updateTaskMock}
        deleteTask={deleteTaskMock}
      />
    );
    fireEvent.click(getByLabelText('trash-icon'));
    expect(deleteTaskMock).toHaveBeenCalled();
  });

  test('should match to snapshot', () => {
    const openModalMock = vi.fn();
    const updateTaskMock = vi.fn();
    const deleteTaskMock = vi.fn();
    const { container } = render(
      <Card
        title="Test Title"
        openModal={openModalMock}
        updateTask={updateTaskMock}
        deleteTask={deleteTaskMock}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
