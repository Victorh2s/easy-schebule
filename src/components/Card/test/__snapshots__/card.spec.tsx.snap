// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card component > should match to snapshot 1`] = `
.c0 {
  height: 10vh !important;
  border-radius: 0.5rem !important;
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #333333;
  overflow-y: hidden !important;
}

.c0 h4 {
  width: 100%;
  height: 5vh;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  cursor: pointer;
}

.c0 nav {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-left: 1rem;
}

.c0 nav svg {
  cursor: pointer;
}

.c0 nav svg + svg {
  margin-top: 0.5rem;
}

<div>
  <div
    class="c0"
  >
    <h4>
      Test Title
    </h4>
    <nav>
      <svg
        aria-label="pen-icon"
        fill="currentColor"
        height="20"
        viewBox="0 0 256 256"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M227.32,73.37,182.63,28.69a16,16,0,0,0-22.63,0L36.69,152A15.86,15.86,0,0,0,32,163.31V208a16,16,0,0,0,16,16H92.69A15.86,15.86,0,0,0,104,219.31l83.67-83.66,3.48,13.9-36.8,36.79a8,8,0,0,0,11.31,11.32l40-40a8,8,0,0,0,2.11-7.6l-6.9-27.61L227.32,96A16,16,0,0,0,227.32,73.37ZM48,179.31,76.69,208H48Zm48,25.38L51.31,160,136,75.31,180.69,120Zm96-96L147.32,64l24-24L216,84.69Z"
        />
      </svg>
      <svg
        aria-label="trash-icon"
        fill="currentColor"
        height="20"
        viewBox="0 0 256 256"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M216,48H176V40a24,24,0,0,0-24-24H104A24,24,0,0,0,80,40v8H40a8,8,0,0,0,0,16h8V208a16,16,0,0,0,16,16H192a16,16,0,0,0,16-16V64h8a8,8,0,0,0,0-16ZM96,40a8,8,0,0,1,8-8h48a8,8,0,0,1,8,8v8H96Zm96,168H64V64H192ZM112,104v64a8,8,0,0,1-16,0V104a8,8,0,0,1,16,0Zm48,0v64a8,8,0,0,1-16,0V104a8,8,0,0,1,16,0Z"
        />
      </svg>
    </nav>
  </div>
</div>
`;
